src/servant/tt_003/demon.c

```diff
@@ -229,7 +229,168 @@ INCLUDE_ASM("servant/tt_003/nonmatchings/demon", func_us_80172EF8);
+        cosine = self->ext.et_801737F0.stepCounter % 10;
+        if ((cosine == 0) &&
+            (self->ext.et_801737F0.animationTriggerCount < 2)) {
+            self->ext.et_801737F0.animationTriggerCount++;
+        }
+
+        self->ext.et_801737F0.animationTimer += 8;
+
+        if ((s16)self->ext.et_801737F0.animationTimer >= 0x100) {
+            self->step += 1;
+        }
+        break;
+
+    case 2:
+        self->ext.et_801737F0.stepCounter++;
+
+        if ((s16)self->ext.et_801737F0.stepCounter >= 51) {
+            CreateEventEntity(self, 0xDD, 0);
+            DestroyEntity(self);
+            return;
+        }
+        break;
+
+    default:
+
+        break;
+    }
+
+    self->ext.et_801737F0.angle += 0x400;
+    self->ext.et_801737F0.angle &= 0xFFF;
+
+    sine = rsin(self->ext.et_801737F0.angle);
+    cosine = rcos(self->ext.et_801737F0.angle);
+
+    posXOffset = (self->posX.i.hi) + ((self->facingLeft) ? (6) : (-6));
+    posYOffset = self->posY.i.hi - 0xC;
+    prim = &g_PrimBuf[self->primIndex];
+    scaledCounter = self->ext.et_801737F0.animationTimer / 16;
+    prim->x0 = posXOffset +
+               ((cosine * -(scaledCounter)-sine * -(scaledCounter)) >> 12);
+    prim->y0 = posYOffset +
+               ((sine * -(scaledCounter) + cosine * -(scaledCounter)) >> 12);
+
+    prim->x1 =
+        posXOffset + ((cosine * (scaledCounter)-sine * -(scaledCounter)) >> 12);
+    prim->y1 = posYOffset +
+               ((sine * (scaledCounter) + cosine * -(scaledCounter)) >> 12);
+
+    prim->x2 =
+        posXOffset + ((cosine * -(scaledCounter)-sine * (scaledCounter)) >> 12);
+    prim->y2 = posYOffset +
+               ((sine * -(scaledCounter) + cosine * (scaledCounter)) >> 12);
+
+    prim->x3 =
+        posXOffset + ((cosine * (scaledCounter)-sine * (scaledCounter)) >> 12);
+    prim->y3 = posYOffset +
+               ((sine * (scaledCounter) + cosine * (scaledCounter)) >> 12);
+
+    colorVal = ((self->ext.et_801737F0.stepCounter & 1) << 7) & 0xFF;
+
+    colorVal = ((self->ext.et_801737F0.stepCounter & 1) << 7) & 0xFF;
+    PCOL(prim) = colorVal;
+
+    prim = prim->next;
+
+    PCOL(prim) = self->ext.et_801737F0.animationTimer / 2;
+
+    inverseCounter = 0x100 - self->ext.et_801737F0.animationTimer;
+    if (inverseCounter < 0) {
+        inverseCounter += 7;
+    }
+    newX = posXOffset - (inverseCounter >> 3);
+    prim->x0 = prim->x2 = newX;
+
+    inverseCounter = 0x100 - self->ext.et_801737F0.animationTimer;
+    if (inverseCounter < 0) {
+        inverseCounter += 7;
+    }
+    newX = posXOffset + (inverseCounter >> 3);
+    prim->x1 = prim->x3 = newX;
+
+    inverseCounter = 0x100 - self->ext.et_801737F0.animationTimer;
+    if (inverseCounter < 0) {
+        inverseCounter += 7;
+    }
+    newY = posYOffset - (inverseCounter >> 3);
+    prim->y0 = prim->y1 = newY;
+
+    inverseCounter = 0x100 - self->ext.et_801737F0.animationTimer;
+    if (inverseCounter < 0) {
+        inverseCounter += 7;
+    }
```

Comment:
I'm pretty sure these are divisions, did not spend too much time playing around with it but this is a closer match for PSP (but farther from PSX)
https://decomp.me/scratch/RWU4x